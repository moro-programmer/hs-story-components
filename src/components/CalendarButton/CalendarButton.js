import React, { Component, PropTypes } from 'react';
import Moment from 'moment';
import classnames from 'classnames';
import './CalendarButton.scss';

class CalendarButton extends Component {
    render() {
        const {active, onClick, className, children} = this.props;
        const {header, text} = this.getData();

        return (
            <div {...this.props}
                 onClick={onClick}
                 className={classnames(className, 'hs-calendar-button', {'hs-calendar-button--active': active})}>
                <div className='hs-calendar-button__header'>{header.toUpperCase()}</div>
                <div className='hs-calendar-button__text'>{text}</div>
                {children}
            </div>
        );
    }

    getData() {
        let {header = '', text, date, dateTextFormat, dateHeaderFormat} = this.props;
        let result;

        if (date) {
            date = new Moment(date);
            result = {
                header: date.format(dateHeaderFormat),
                text: date.format(dateTextFormat)
            };
        } else {
            result = {header, text};
        }

        return result;
    }
}

CalendarButton.propTypes = {
    header: PropTypes.string,
    text: PropTypes.string,
    active: PropTypes.bool,
    className: PropTypes.string,
    date: PropTypes.object,
    dateTextFormat: PropTypes.string,
    dateHeaderFormat: PropTypes.string,
    onClick: PropTypes.func,
    children: PropTypes.node
};

CalendarButton.defaultProps = {
    active: false,
    dateTextFormat: 'D',
    dateHeaderFormat: 'MMM'
};

export default CalendarButton;