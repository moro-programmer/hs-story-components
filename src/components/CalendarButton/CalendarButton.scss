$active-color: #0099cc;
$active-color-hover: #86cdef;
$inactive-color: #95989a;
$inactive-color-hover: #c6c6c6;

.hs-calendar-button {
    font-family: Roboto, Verdana, Arial, sans-serif;
    height: 40px;
    width: 40px;
    border: 1px solid $inactive-color;
    background: $inactive-color-hover;
    text-align: center;
    font-weight: 700;
    cursor: pointer;
    position: relative;

    &:hover {
        border: 1px solid $inactive-color-hover;
        background: $inactive-color-hover;

        .hs-calendar-button__header {
            border-bottom: 1px solid $inactive-color-hover;
        }
        .hs-calendar-button__text {
            color: $inactive-color-hover;
        }
    }

    .hs-calendar-button__header {
        height: 14px;
        color: #fff;
        font-size: 10px;
        border-bottom: 1px solid $inactive-color;
    }

    .hs-calendar-button__text {
        font-size: 15px;
        height: 24px;
        line-height: 24px;
        color: $inactive-color;
        background: #fff;
    }

    &.availability-add-new {
        .hs-calendar-button__text {
            font-size: 17px;
        }
    }
}

.hs-calendar-button--active {
    border: 1px solid $active-color;
    background: $active-color;

    &:hover {
        border: 1px solid $active-color-hover;
        background: $active-color-hover;

        .hs-calendar-button__header {
            border-bottom: 1px solid $active-color-hover;
        }

        .hs-calendar-button__text {
            color: $active-color-hover;
        }
    }

    .hs-calendar-button__header {
        border-bottom: 1px solid $active-color;
    }
    .hs-calendar-button__text {
        color: $active-color;
    }
}
