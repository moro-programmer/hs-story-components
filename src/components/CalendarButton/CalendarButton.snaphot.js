import React from 'react';
import CalendarButton from './CalendarButton.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<CalendarButton header='---' text='+'/>).toJSON();
  expect(tree).toMatchSnapshot();
});
