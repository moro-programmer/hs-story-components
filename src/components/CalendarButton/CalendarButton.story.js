import React from 'react';
import { storiesOf } from '@kadira/storybook';
import CalendarButton from './CalendarButton';
import { withKnobs, text, boolean, number } from '@kadira/storybook-addon-knobs';

storiesOf('CalendarButton', module)
    .addDecorator(withKnobs)
    .addWithInfo('active CalendarButton', () => (
          <CalendarButton active={boolean('active',true)} header='SEP' text='7'/>
    ),{ inline: true })
    .add('inactive CalendarButton',()=>{
          <CalendarButton header='SEP' text='7'/>
    })
    .add('CalendarButton without header',()=>{
          <CalendarButton text='+'/>
    });
