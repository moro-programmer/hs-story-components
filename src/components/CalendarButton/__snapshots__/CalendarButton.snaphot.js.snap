exports[`test renders correctly 1`] = `
<div
  className="hs-calendar-button"
  dateHeaderFormat="MMM"
  dateTextFormat="D"
  header="---"
  onClick={undefined}
  text="+">
  <div
    className="hs-calendar-button__header">
    ---
  </div>
  <div
    className="hs-calendar-button__text">
    +
  </div>
</div>
`;
